web/src/features/fbaCalculator/RowManager.ts

```diff
@@ -50,84 +50,99 @@ export class RowManager {
     tableRows: FBATableRow[]
   ): FBATableRow[] => {
     const reverseOrder = order === 'asc' ? 'desc' : 'asc'
+    /**
+     * Partitions table rows into non-empty and empty row arrays, sorts the non-empty row array,
+     * then concatenates the empty row array to the end of the sorted, non-empty row array.
+     *
+     * @param field FBATableRow field to order by
+     * @param otherOrder optional Order ordering should follow, otherwise the current order
+     * @returns rows sorted by the specified field, with empty rows always at the end
+     */
+    const orderByWithEmptyAtBottom = (field: string, otherOrder?: Order) => {
+      const partition = _.partition(tableRows, row => !!_.get(row, field, null))
+      return _.concat(
+        _.orderBy(partition[0], field, otherOrder ? otherOrder : order),
+        partition[1]
+      )
+    }
     switch (sortByColumn) {
       case SortByColumn.Zone: {
-        return _.orderBy(tableRows, 'zone_code', order)
+        return orderByWithEmptyAtBottom('zone_code')
```

Comment:
hmmm I recommend double-checking with @tessmgood on this - I know BB didn't like the behaviour that new rows automatically go to the bottom as soon as the "Add Row" button is clicked, because when you have a long list of stations, it means scrolling up to the top to click the add row button, then scrolling to the bottom of the page to add the station info, etc. And repeat.

I think the reason the ticket was created was because it was illogical that when sorting by HFI descending, rows with no HFI data were being listed before the rows with high HFIs.

Reply:
Good point - I'm looking at BB's feedback as a separate ticket to isolate the interaction of adding a new row (it's different to sorting behaviour IMO) - I have thoughts & ideas about this that I will put in a ticket.